### ISSUE TITLE
[Discuss on consistency] Custom data on Realsense L515+ORB-SLAM3+Open3D Reconstruction

### ISSUE THREAD

**Issue by glennliu** [hustvl/HAIS#33]
Hi Shaoyun

Thanks for your work. I'm interested in how HAIS can help the SLAM system extract persistent semantic landmarks. 

I have tested HAIS on my own dataset,
- Hardware: Intel Realsense L515 RGB-D camera
- Localization: ORB-SLAM3 with purely RGB-D input
- Reconstruction: Open3D RGB-D integration, which is based on TSDF and extract point cloud after the entire scan is finished. 

Here are some results (colored by raw RGB and semantic segmentation),
a. Living Room
![living_mesh](https://user-images.githubusercontent.com/8009679/196634543-2d327b66-6c67-464b-bf37-35efb4a04f65.png)

![living](https://user-images.githubusercontent.com/8009679/196634626-363ffa46-a127-4a7e-8271-b8fe1b0f7656.png)

b. dinning room
![dinning_mesh](https://user-images.githubusercontent.com/8009679/196635013-5fef83c8-395f-4639-be29-54c443ff816d.png)

![dinning](https://user-images.githubusercontent.com/8009679/196635045-88829044-a322-4812-b176-63dad67e7000.png)

c. study room
![studyroom_mesh](https://user-images.githubusercontent.com/8009679/196635155-d31da6b2-5165-420b-985c-d73cba8bb45f.png)

![studyroom](https://user-images.githubusercontent.com/8009679/196635167-49e67d97-9a03-43c8-9f06-8b172fc53f4f.png)

The quantitative results show there is quite many over-segmentation. In the ScanNet test results, I can also see a few of the over-segmentation problems. But it is way less frequent than in my customized dataset. Moreover, over-segmentation in ScanNet dataset normally occurs in those poorer reconstructed sub-volumes, while it occurs in well-reconstructed spaces in the customized data, such as the _Living Room_ scan.

So, what is actually affecting the segmentation performance in the self-collected dataset? 
- Previous issues #19 also discuss the issues. And some says the ```class_numpoint_mean_dict``` should be modified. But I'm running in indoor scenes similar to ScanNet. ```mean_numpoint``` and ```mean_radius``` should be only slightly different. That dictionary parameter should not affect the consistency across datasets in my case. 
- How is the domain shift of HAIS? What kind of issues can affects the consistency across different dataset?

Thanks
Chuhao

**Comment by glennliu** [comment]
I also runs it on several SceneNN dataset. 
![bedroomnn](https://user-images.githubusercontent.com/8009679/196641042-1873a22c-81c2-46a1-b038-18a66852848c.png)

![bedroomnn_semantic](https://user-images.githubusercontent.com/8009679/196641099-b2433a47-8e7b-4082-9a70-5d673133e18c.png)
The scene is also constructed by Open3D integration, but using the ground-truth pose.

I can also see some over-segmentation, such as the broken chair and table. 
I understand this is a long-term challenge in scene segmentation tasks and cannot be perfectly solved. But I just want to discuss what affects it and how to improve it across different dataset. 

Thanks

**Comment by outsidercsy** [comment]
Thanks for the interesting question.  Over-seg. and under-seg. are related to the instance seg. but not semantic seg. Your visualization reflects that class labels of some points are wrongly predicted, thus the prolem lies in the first stage. HAIS is trained on ScanNetV2 whose data amount is far away from enough. I think the data amount accounts for the wrong prediction and maybe adopts more data can alleviate the problem. And there do exist gaps between datasets, e.g., point cloud density. Some parameters may bu tuned for better results in other datasets.
Hope these help you.

**Comment by glennliu** [comment]
@outsidercsy Thanks for your time. 
I agree with your points that data amount is one of the reasons. However, the datasets are already close to ScanNet data.
Those scene I choose are very similiar to some scenes in ScanNet (indoor living room etc.). And ScanNet is reconstructed using TSDF mapping, which is the same as what I used and SceneNN used. The RGB-D cameras we used are different but should not be too much difference.Besides, HAIS is trained with 1200 scans from ScanNet which is not a small amount. So, I was expecting very similar performance as ScanNet.
Overall, I think the data amount does matter. But there should be a more general method to improve the consistency of the segmentation network.

**Comment by outsidercsy** [comment]
I mean that the data amount of ScanNet is also not enough for convergence. And I also observe many cases with poor segmentation results on ScanNet. What if pretraining on ScanNet and then finetuning on your dataset?

**Comment by glennliu** [comment]
Some subvolumes in ScanNet are also reconstructed poorly. It should be reasonable that those poor volumes in ScanNet cannot be well segmented. I'd like to try finetuning my dataset. Do you have any suggestion on which parts of the parameter should be considered in finetuning?

**Comment by glennliu** [comment]
Here are more results (raw rgb and instance segmentation+semantic text)
_living_room_
![livingb_mesh](https://user-images.githubusercontent.com/8009679/197316395-aaed4ebe-2346-49e7-a313-1fca42d22810.png)
![livingb](https://user-images.githubusercontent.com/8009679/197316400-881b152c-6a39-4757-a662-d813e6d6ce17.png)

_washing_room_
![washing_mesh](https://user-images.githubusercontent.com/8009679/197316429-cac34ad2-0bc2-4e7d-aab6-181fd4e7dfb1.png)
![washing](https://user-images.githubusercontent.com/8009679/197316456-8fa1519f-3f6a-4e42-9eb6-302ed449e998.png)

_bedroom_
![bedroom_mesh](https://user-images.githubusercontent.com/8009679/197316463-de13c987-8c59-4fc8-8082-e8b986f50767.png)
![bedroom](https://user-images.githubusercontent.com/8009679/197316468-1f441e31-b71d-4970-8bb8-0f4bb7e90b3f.png)

Beyond over-segmentation, there are a few falsely predicted semantic labels.

**Comment by outsidercsy** [comment]
Thanks for the visualizaiton.  As for data amount, ScanNet200 is released with much more annotated scenes and instances.  More training data helps to solve the bad cases.  https://rozdavid.github.io/scannet200
